apps/api/src/domain/models/context/text-field-context/text-field-context.entity.ts

```diff
@@ -0,0 +1,36 @@
+import { PartialDTO } from 'apps/api/src/types/partial-dto';
+import { EdgeConnectionContext } from '../context.entity';
+import { EdgeConnectionContextType } from '../types/EdgeConnectionContextType';
+
+export class TextFieldContext extends EdgeConnectionContext {
+    readonly type = EdgeConnectionContextType.textField;
+
+    // This is the key (field name) of the target text property on the model
+    readonly target: string;
+
+    /**
+     * The range of characters to use for the context.
+     *
+     * TODO [design] How should we represent the entire string as context?
+     * We'll worry about this when we hit the use case
+     * Ideas:
+     * - Use [o,s.length] for the entire text field to be used as context.
```

Comment:
Not sure I understand the expression `[o,s.length]`

Reply:
I reworded this.